packages/backend/src/peripherals/database/ReportRepository.ts

```diff
@@ -60,16 +80,59 @@ export class ReportRepository extends BaseRepository {
     return await knex('reports').delete()
   }
 
+  private getByProjectAndAssetQuery(
+    knex: Knex,
+    projectId: ProjectId,
+    assetId: AssetId,
+  ) {
+    return knex('reports')
+      .where('asset_id', assetId.toString())
+      .andWhere('project_id', projectId.toString())
+      .orderBy('unix_timestamp')
+  }
+
   async getDailyByProjectAndAsset(
     projectId: ProjectId,
     assetId: AssetId,
   ): Promise<ReportRecord[]> {
     const knex = await this.knex()
-    const rows = await knex('reports')
-      .where('asset_id', assetId.toString())
-      .andWhere('project_id', projectId.toString())
-      .andWhere('is_daily', true)
-      .orderBy('unix_timestamp', 'asc')
+    const rows = await this.getByProjectAndAssetQuery(
+      knex,
+      projectId,
+      assetId,
+    ).andWhere('is_daily', true)
+    return rows.map(toRecord)
+  }
+
+  async getHourlyByProjectAndAsset(
+    projectId: ProjectId,
+    assetId: AssetId,
+  ): Promise<ReportRecord[]> {
+    const knex = await this.knex()
+    const rows = await this.getByProjectAndAssetQuery(
+      knex,
+      projectId,
+      assetId,
+    ).andWhere(
+      'unix_timestamp',
+      '>=',
+      UnixTime.now().add(-7, 'days').toString(),
```

Comment:
This is part of business logic and arguably makes this query less testable. Could we instead pass this as a function argument? Same goes for sixHourly and the other repository.